src/modules/library.ts

```diff
@@ -26,303 +33,605 @@ export enum Buttons {
   }
 
   /** Verifies the specified channel is installed on the device. */
-  async verifyIsChannelExist({ apps, id, retries = 3 }: { apps?: appResponse[]; id: string; retries?: number }) {
+  async verifyIsChannelExist({ apps, id, retries }: { apps?: Apps; id: string; retries?: number }) {
     if (!apps) {
       apps = await this.getApps(retries);
     }
-    return !!apps.find(app => app.ID === id);
+    for (const key of Object.keys(apps)) {
+      if (apps[key].id === id) {
+        return true;
+      }
+    }
+    return false;
   }
 
   /** Verify that the screen is loaded based on the provided element data. */
   async verifyIsScreenLoaded({
     data,
-    maxRetries = 10,
-    delayInMillis = 1000,
-    postRetries = 3,
+    maxAttempts = 10,
+    delayInMillis = this.retryDelayInMillis,
+    httpRetries = this.retries,
   }: {
-    data: elementDataObject;
-    maxRetries?: number;
+    data: ElementDataObject;
+    maxAttempts?: number;
     delayInMillis?: number;
-    postRetries?: number;
+    httpRetries?: number;
   }) {
-    let functionRetries = 0;
-    while (functionRetries < maxRetries) {
-      const uiLayoutResponse = await this.driver.getUIElementError({ data, retries: postRetries });
-      if (uiLayoutResponse.status !== 200) functionRetries++;
-      else return true;
-      await sleep(delayInMillis);
+    let attempts = 0;
+    while (attempts < maxAttempts) {
+      try {
+        const response = await this.driver.getUIElement({ data, retries: httpRetries });
+        if (response) {
+          return true;
+        } else {
+          attempts++;
+          await sleep(delayInMillis);
+        }
+      } catch (error) {
+        console.log(error);
+        attempts++;
+        await sleep(delayInMillis);
+      }
     }
     return false;
   }
 
   /** Alias for verifyIsScreenLoaded with a more intuitive name */
-  async verifyIsElementOnScreen({
+  verifyIsElementOnScreen({
     data,
-    maxRetries = 10,
-    delayInMillis = 1000,
-    postRetries = 3,
+    maxAttempts = 10,
+    delayInMillis = this.retryDelayInMillis,
+    httpRetries = this.retries,
   }: {
-    data: elementDataObject;
-    maxRetries?: number;
+    data: ElementDataObject;
+    maxAttempts?: number;
     delayInMillis?: number;
-    postRetries?: number;
+    httpRetries?: number;
   }) {
-    return this.verifyIsScreenLoaded({ data, maxRetries, delayInMillis, postRetries });
+    return this.verifyIsScreenLoaded({ data, maxAttempts, delayInMillis, httpRetries });
   }
 
   /** Simulates the press and release of the specified key. */
   async pressBtn({
     keyPress,
-    delayInMillis = 2000,
-    retries = 3,
+    delayInMillis = this.pressDelayInMillis,
+    retries = this.retries,
+    params,
   }: {
     keyPress: string;
     delayInMillis?: number;
-    retries?: 3;
+    retries?: number;
+    params?: Params;
   }) {
+    const response = this.driver.sendKeypress({ keyPress, retries, params });
```

Comment:
Is this not an async call?

Reply:
Good catch! My next commit will have that changed.

Reply:
c871193